### PR TITLE
Codegpt, env variables for which gpu to use, parallel inference

### PR DESCRIPTION
**Pull request by timvandam**
This is already running on the server (and works)

### CODE REVIEW

File: code4me-server/src/api.py [hunk 1]
```diff
@@ -45,8 +47,15 @@ def autocomplete():
     t_before = datetime.now()
     predictions = {}
     unique_predictions_set = set()
-    for model in Model:
-        model_predictions = model.value[1](left_context, right_context)
+
+    def predict_model(model: Model) -> List[str]:
+        start_time = time.time()
+        res = model.value[1](left_context, right_context)
+        print(f"Model {model.name} took {time.time() - start_time} seconds", flush=True)
```
[the remark this answers is not in the document]
Comment (timvandam) [reply code4me-server/src/api.py:54]: :heavy_check_mark:

File: code4me-server/src/codegpt.py [hunk 2]
```diff
@@ -197,8 +208,10 @@ def codegpt_predict(left_context: str, right_context: str):
...
-            return text
-    return ""
-
+            if tokenizer.eos_token_id in t:
+                t = t[:t.index(tokenizer.eos_token_id)]
+            if tokenizer.sep_token_id in t:
+                t = t[:t.index(tokenizer.sep_token_id)]
+            text = DecodeIds(t).strip()
+            return [text]
+    return []
```
[the remark this answers is not in the document]
Comment (timvandam) [reply code4me-server/src/codegpt.py:217]: :ballot_box_with_check:

File: docker-compose.yml [hunk 3]
```diff
@@ -31,8 +31,13 @@ services:
...
       - ./code4me-server/templates:/codeforme/templates
       - ./code4me-server/static:/codeforme/static
+      - ./codegpt_checkpoint:/codeforme/codegpt_checkpoint
     expose:
       - 3000
     environment:
       - CODE4ME_CUDA=True
       - "SURVEY_LINK=https://docs.google.com/forms/d/1uES5o6etbWEZVNpUc0TGisDJXIIuj5hIwa9tF7_FQuw/?entry.1566855902={user_id}"
+      - CODEGPT_CHECKPOINT_PATH=/codeforme/codegpt_checkpoint
+      - UNIXCODER_DEVICE=cuda:0
```
[the remark this answers is not in the document]
Comment (timvandam) [reply docker-compose.yml:41]: Done. Testing

**Comment by timvandam** [comment]
@FrankHeijden did your changes

**Review by FrankHeijden — APPROVED** [review]

**Pull request merged**